clippy_lints/src/await_holding_invalid.rs

```diff
@@ -173,7 +174,7 @@ declare_clippy_lint! {
 impl_lint_pass!(AwaitHolding => [AWAIT_HOLDING_LOCK, AWAIT_HOLDING_REFCELL_REF, AWAIT_HOLDING_INVALID_TYPE]);
 
 pub struct AwaitHolding {
-    def_ids: DefIdMap<(&'static str, Option<&'static str>)>,
+    def_ids: DefIdMap<(&'static str, &'static DisallowedPath)>,
```

Comment:
I don't think we should allow specifying a replacement here, even if it's never used. Also the reasoning should usually be inferred due to it being a specific lint, right?

Reply:
I see your point. I'll work on a fix.

Reply:
My solution may be over-engineered. Other alternatives I considered were:
- Use [`serde-untagged`](https://github.com/dtolnay/serde-untagged), but it is not in the list of [permitted dependencies](https://github.com/rust-lang/rust/blob/8adb4b30f40e6fbd21dc1ba26c3301c7eeb6de3c/src/tools/tidy/src/deps.rs#L236).
- Implement a custom visitor like `serde-untagged` does, but that seemed yet more complicated.
- Don't error during deserialization and instead warn when the replacement is discovered.

I'm open to suggestions if another approach is desired.